R/utility_spliceHelper_functions.R

```diff
@@ -1,598 +1,572 @@
+                                                               subjectEndRng)
+  annotatedTable$alternativeTES <- calculateTESdistance(queryEndRng, 
+                                                       subjectEndRng, 
+                                                       annotatedTable$alternativeLastExon,
+                                                       strand)
+  annotatedTable$internalLastExon.query <- annotateInternalStartEnd(queryEndRng,
+                                                                   subjectFullRng, annotatedTable$alternativeLastExon)
+  annotatedTable$internalLastExon.subject <- annotateInternalStartEnd(subjectEndRng,
+                                                                     queryFullRng, annotatedTable$alternativeLastExon)
+  #intron retention subject and query
+  annotatedTable$intronRetention.subject <- annotateIntronRetent(querySpliceRng, subjectFullRng)
+  annotatedTable$intronRetention.query <- annotateIntronRetent(subjectSpliceRng ,queryFullRng)
+  #exon skipping query and subject
+  annotatedTable$exonSkipping.query <- annotateExonSkip(querySpliceRng, subjectFullRng,
+                                                        subjectStartRng, subjectEndRng)
+  annotatedTable$exonSkipping.subject <- annotateExonSkip(subjectSpliceRng ,queryFullRng,
+                                                          queryStartRng, queryEndRng) 
+  # exon 3' splice site
+  exonSpliceTable <- annotateExonSplice(querySpliceRng, subjectFullRng,
+                                        subjectStartRng, subjectEndRng,
+                                        annotatedTable$strand)
+  annotatedTable <- cbind(annotatedTable, exonSpliceTable)
+  return(annotatedTable)
+}
 
-  subjectEndExonStartExtension <- start(subjectEnd.querySplice.Rng)<end(qSpRngList) & end(subjectEnd.querySplice.Rng)>end(qSpRngList) & start(qSpRngList)< start(subjectEnd.querySplice.Rng)
-  subjectEndExonStartExtension <- tapply(subjectEndExonStartExtension, subjectStartEnd.querySplice.IdMap, sum)
+#' extract strand from GRangesList
+#' @noRd
+getStrandFromGrList <- function(grl) { 
+  return(unlist(strand(grl), use.names = F)[cumsum(elementNROWS(grl))]) 
+  }
+
+#' start/end ranges pre-processing                                 
+expandRanges <- function(unexpandedRng,targetRng){ 
+  processedRng <- rep(unexpandedRng,elementNROWS(targetRng))
+  mcols(processedRng)$IdMap <- rep(1:length(unexpandedRng),elementNROWS(targetRng))
+  mcols(processedRng)$matchRng <- unlist(targetRng, use.names=F)
+  return (processedRng)
+}
 
-  subjectEndExonEndExtension <- end(subjectEnd.querySplice.Rng)>start(qSpRngList) & start(subjectEnd.querySplice.Rng)<start(qSpRngList) & end(qSpRngList)> end(subjectEnd.querySplice.Rng)
-  subjectEndExonEndExtension <- tapply(subjectEndExonEndExtension, subjectStartEnd.querySplice.IdMap, sum)
+#' splice ranges pre-processing
+expandRangesList <- function(unexpandedRng,targetRng){ 
```

Comment:
also add some comments on what this function does

Reply:
I have added a description to this function.